pkg/sbom/nodelist.go

```diff
@@ -193,13 +194,13 @@ func (nl *NodeList) MergeEdges(es []*Edge) {
 // More than one root element can be added to the NodeList.
 func (nl *NodeList) AddRootNode(n *Node) {
 	if n.Id == "" {
-		// TODO warn here
-		return
+		logrus.Warnf("Node with empty ID, creating a new Id: %s@%s\n", n.Name, n.Version)
+		n.Id = NewNodeIdentifier(fmt.Sprintf("%s@%s", n.Name, n.Version))
 	}
 
 	for _, id := range nl.RootElements {
```

Comment:
We can now change this loop to use the slices package
```suggestion
	if slices.Contains(nl.RootElements, id) {
	   return
	}
```